components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/handlers/TokenBindingExpiryEventHandler.java

```diff
@@ -100,6 +113,124 @@ public void handleEvent(Event event) throws IdentityEventException {
+            SessionContext sessionContext = (SessionContext) eventProperties.get(IdentityEventConstants
+                    .EventProperty.SESSION_CONTEXT);
+            if (sessionContext != null) {
+                Map<String, SequenceConfig> authenticatedSequences = sessionContext.getAuthenticatedSequences();
+                if (MapUtils.isEmpty(authenticatedSequences)) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Could not find the authenticated sequence config map in the session context");
+                    }
+                    return;
+                }
+
+                AuthenticatedUser user = (AuthenticatedUser) sessionContext
+                        .getProperty(FrameworkConstants.AUTHENTICATED_USER);
+                if (user == null) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Authenticated user is nnot founnd in the session context");
+                    }
+                    return;
+                }
+                String tenantDomain = user.getTenantDomain();
+                if (StringUtils.isBlank(tenantDomain)) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Could not find the tenant domain of the user: " + user.getUserName());
+                    }
+                    return;
+                }
+                for (Map.Entry<String, SequenceConfig> sequenceConfigEntry : authenticatedSequences.entrySet()) {
+                    String applicationName = sequenceConfigEntry.getValue().getApplicationId();
+                    String clientId = getClientIdFromApplicationName(applicationName, tenantDomain);
+                    String bindingType = OAuth2Util.getAppInformationByClientId(clientId).getTokenBindingType();
+                    if (OAuth2Constants.TokenBinderType.SSO_SESSION_BASED_TOKEN_BINDER.equals(bindingType)) {
+                        revokeTokensOfBindingRef(user, OAuth2Util.getTokenBindingReference(sessionContextIdentifier));
+                    }
+                }
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug("Session context for session context identifier: " + sessionContextIdentifier +
+                            " is not found in the event");
+                }
+            }
+        }
+    }
+
+    /**
+     * Get session context identifier from the event.
+     *
+     * @param event Event.
+     * @return Session context identifier.
+     */
+    private String getSessionIdentifier(Event event) {
+
+        Map<String, Object> eventProperties = event.getEventProperties();
+        Map<String, Object> paramMap = (Map<String, Object>) eventProperties.get(IdentityEventConstants
+                .EventProperty.PARAMS);
+        String sessionContextIdentifier = null;
+        for (Map.Entry<String, Object> entry : paramMap.entrySet()) {
+            log.info(entry.getKey() + ":" + entry.getValue());
```

Comment:
Does this need to be an info log. can it be a debug log? Add a message before printing the logs.